pkg/apiserver/resthandler.go

```diff
@@ -753,7 +781,7 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
+			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
```

Comment:
Good point, current admission for quota has to look up the old object to
determine which scope it falls under.  This would replace the need for that.

On May 23, 2016, at 8:44 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/apiserver/resthandler.go
<https://github.com/kubernetes/kubernetes/pull/25787#discussion_r64313140>:

> @@ -753,7 +781,7 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
>  		if admit != nil && admit.Handles(admission.Delete) {
>  			userInfo, _ := api.UserFrom(ctx)
>
> -			err = admit.Admit(admission.NewAttributesRecord(nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
> +			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))

just off the top of my head, hints for releasing quota based on the old
object, the ability to have admission limit access to objects with certain
fields, etc

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/25787/files/bf47e5790193381752c396876a2ce9950fe01e6d#r64313140>